### PR TITLE
Add replacement add-ons endpoint for taar to use.

### LINKED ISSUE #6495: Add JSON export of replacement data
The TAAR service needs data from the replacements table. We currently have a mapping of guid -> path. The data TAAR needs is guid -> guid.

Assuming getting at the data isn't too problematic it would ideal if there was a way to export the source / replacement guid data as JSON. This could either be an export in the admin or as an API.

From talking to @jvillalobos I don't think there's a reason that we can't make the replacement data public.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #6495

### CODE REVIEW

**Comment by eviljeff** [comment]
@Dexterp37 @mlopatka

File: src/olympia/addons/serializers.py [hunk 1]
```diff
@@ -591,3 +596,46 @@ class Meta:
...
+        try:
+            get_args = {'slug': collection_slug, 'listed': True}
+            if isinstance(user_id, basestring) and not user_id.isdigit():
+                get_args.update(**{'author__username': user_id})
+            else:
+                get_args.update(**{'author': user_id})
+            collection = Collection.objects.get(**get_args)
+        except Collection.DoesNotExist:
+            return []
+        valid_q = Addon.objects.get_queryset().valid_q()
```
Comment (diox) [inline src/olympia/addons/serializers.py]: I was confused initially and thought this allowed non-public add-ons to be returned... IMHO `valid_q()` should not have a default add-on status and all callers should provide one...

Comment (eviljeff) [reply src/olympia/addons/serializers.py]: yeah... everywhere else it's used (now, anyway) supplies statuses.  I'll add one.

Comment (eviljeff) [reply src/olympia/addons/serializers.py]: Also, once old collections views are deleted (post frontend) the function can be simplified a lot - the `prefix` parameter is only used in `collection_detail`

**Pull request merged**